feat: add C implementation for `math/base/special/bessely1`

ShabiShett07 opened the pull request:
Progresses #649 

## Description

> What is the purpose of this pull request?

This pull request:

-   progresses #649  https://github.com/stdlib-js/stdlib/issues/649 by adding math/base/special/bessely1
## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses #649 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.
Open to reviews!!

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

ShabiShett07 committed:
feat: add math/base/special/bessely1

ShabiShett07 committed:
feat: add math/base/special/bessely1

ShabiShett07 commented:
Errors have been fixed

ShabiShett07 commented:
/stdlib check-files

ShabiShett07 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

ShabiShett07 commented:
/stdlib lint-autofix

stdlib-bot committed:
fix: resolve lint errors

ShabiShett07 committed:
Fixed package.json

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 commented:
I have fixed the errors in package.json

ShabiShett07 committed:
feat: add /blas/base/zrotg

stdlib-bot committed:
chore: update copyright years

ShabiShett07 commented:
@Neerajpathak07, need your assistance in this!

Neerajpathak07 commented on lib/node_modules/@stdlib/math/base/special/bessely1/README.md:
````diff
@@ -105,19 +105,97 @@ for ( i = 0; i < 100; i++ ) {
--   <span class="package-name">[`@stdlib/math/base/special/besselj0`][@stdlib/math/base/special/besselj0]</span><span class="delimiter">: </span><span class="description">compute the Bessel function of the first kind of order zero.</span>
--   <span class="package-name">[`@stdlib/math/base/special/besselj1`][@stdlib/math/base/special/besselj1]</span><span class="delimiter">: </span><span class="description">compute the Bessel function of the first kind of order one.</span>
--   <span class="package-name">[`@stdlib/math/base/special/bessely0`][@stdlib/math/base/special/bessely0]</span><span class="delimiter">: </span><span class="description">compute the Bessel function of the second kind of order zero.</span>
+```c
+double out = stdlib_base_bessely1( 0.0 );
+// returns -Infinity
+
+out = stdlib_base_bessely1( 1.0 );
+// returns ~-0.781
+```
````
```suggestion
```c
double out = stdlib_base_bessely1( 0.0 );
// returns -Infinity
```
```

Neerajpathak07 commented on lib/node_modules/@stdlib/math/base/special/bessely1/README.md:
```diff
@@ -131,12 +209,6 @@ for ( i = 0; i < 100; i++ ) {
 
 <!-- <related-links> -->
 
-[@stdlib/math/base/special/besselj0]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/besselj0
```
Kindly revert this change

Neerajpathak07 commented on lib/node_modules/@stdlib/math/base/special/bessely1/README.md:
```diff
@@ -105,19 +105,97 @@ for ( i = 0; i < 100; i++ ) {
+
+Computes the [Bessel function of the second kind][bessel-second-kind] of order one at `x`.
 
--   <span class="package-name">[`@stdlib/math/base/special/besselj0`][@stdlib/math/base/special/besselj0]</span><span class="delimiter">: </span><span class="description">compute the Bessel function of the first kind of order zero.</span>
```
revert this change in L211.

Neerajpathak07 replied:
```
## See Also 
```
tag should be lined up above this while respecting the one line space between them.

Neerajpathak07 commented on lib/node_modules/@stdlib/math/base/special/bessely1/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,60 @@
+
+// VARIABLES //
+
+var y0 = tryRequire( resolve( __dirname, './../lib/native.js' ) );
+var opts = {
+	'skip': ( y0 instanceof Error )
+};
```
```suggestion
var y1 = tryRequire( resolve( __dirname, './../lib/native.js' ) );
var opts = {
	'skip': ( y1 instanceof Error )
};
```
According to `benchmark.js`

Neerajpathak07 replied:
Copy paste error from `bessely0`

Neerajpathak07 commented on lib/node_modules/@stdlib/math/base/special/bessely1/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,60 @@
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		x = ( randu() * 100000.0 ) - 0.0;
+		y = y0( x );
```
```suggestion
		y = y1( x );
```
since it is of order 1.

Neerajpathak07 commented on lib/node_modules/@stdlib/math/base/special/bessely1/src/main.c:
```diff
@@ -0,0 +1,307 @@
+/**
+* Computes the Bessel function of the second kind of order one.
+*
+* @param x - input value
+* @returns evaluated Bessel function
```
I believe the C doxygen are incorrect

Neerajpathak07 replied:
```suggestion
* @param x    input value
* @return     evaluated Bessel function
```

Neerajpathak07 commented:
@ShabiShett07 few changes and some additional points:-
- Were the rational functions automatically generated upon running the the corresponding JS file if yes then there's no reason to worry.
- Since this PR aimed at adding C implementation for the existing JS implementation change the PR tittle to:-
 feat: add C implementation for `math/base/special/bessely1`

ShabiShett07 commented:
Thank you @Neerajpathak07 for recognizing the errors, actually this was a PR made by me in initial stages on Stdlib, surely gonna make all the necessary changes.

ShabiShett07 committed:
fix: multiple errors

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

ShabiShett07 commented:
@Neerajpathak07, made all necessary changes as per your suggestion, is there any other errors I need to fix?

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/bessely1/examples/c/example.c:
```diff
@@ -0,0 +1,31 @@
+* limitations under the License.
+*/
+
+#include <stdio.h>
+#include "stdlib/math/base/special/bessely1.h"
```
```suggestion
#include "stdlib/math/base/special/bessely1.h"
#include <stdio.h>
```

gunjjoshi replied:
Let's reorder this so that it matches with the C example in `README.md`.

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/bessely1/README.md:
```diff
@@ -117,7 +205,7 @@ for ( i = 0; i < 100; i++ ) {
 -   <span class="package-name">[`@stdlib/math/base/special/besselj0`][@stdlib/math/base/special/besselj0]</span><span class="delimiter">: </span><span class="description">compute the Bessel function of the first kind of order zero.</span>
 -   <span class="package-name">[`@stdlib/math/base/special/besselj1`][@stdlib/math/base/special/besselj1]</span><span class="delimiter">: </span><span class="description">compute the Bessel function of the first kind of order one.</span>
--   <span class="package-name">[`@stdlib/math/base/special/bessely0`][@stdlib/math/base/special/bessely0]</span><span class="delimiter">: </span><span class="description">compute the Bessel function of the second kind of order zero.</span>
+-   <span class="package-name">[`@stdlib/math/base/special/bessely1`][@stdlib/math/base/special/bessely1]</span><span class="delimiter">: </span><span class="description">compute the Bessel function of the second kind of order one.</span>
```
You can revert this change. We shouldn't modify the `related` sections manually.

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/bessely1/README.md:
```diff
@@ -135,7 +223,7 @@ for ( i = 0; i < 100; i++ ) {
 [@stdlib/math/base/special/besselj1]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/besselj1
 
-[@stdlib/math/base/special/bessely0]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/bessely0
+[@stdlib/math/base/special/bessely1]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/bessely1
```
Similarly, you can revert this too.

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/bessely1/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,60 @@
+
+var resolve = require( 'path' ).resolve;
+var bench = require( '@stdlib/bench' );
+var randu = require( '@stdlib/random/base/randu' );
```
@ShabiShett07 Can you please use [`@stdlib/random/array/uniform`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/random/array/uniform) here?
For reference use, see: https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/acsc/benchmark/benchmark.native.js

gunjjoshi replied:
Would be great if we can reflect this change in `benchmark.js` too.

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/bessely1/lib/native.js:
````diff
@@ -0,0 +1,74 @@
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* ## Notice
+*
+* The original C++ code and copyright notice are from the [Boost library]{@link https://github.com/boostorg/math/blob/develop/include/boost/math/special_functions/detail/bessel_y1.hpp}. The implementation has been modified for JavaScript.
+*
+* ```text
+* Copyright Xiaogang Zhang, 2006.
+*
+* Use, modification and distribution are subject to the
+* Boost Software License, Version 1.0. (See accompanying file
+* LICENSE or copy at http://www.boost.org/LICENSE_1_0.txt)
+* ```
````
```suggestion
```

gunjjoshi replied:
As `native.js` doesn't contain any implementation, we can refrain from adding the reference copyright license.

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/bessely1/src/main.c:
```diff
@@ -0,0 +1,292 @@
+
+// MODULES
+
+#include "stdlib/math/base/special/besselj1.h"
+#include "stdlib/math/base/special/bessely1.h"
```
```suggestion
#include "stdlib/math/base/special/bessely1.h"
#include "stdlib/math/base/special/besselj1.h"
```

gunjjoshi replied:
As this is `bessely1`, let's keep its include path at the top.

ShabiShett07 committed:
docs: add files

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

stdlib-bot committed:
chore: update copyright years

gunjjoshi commented:
@ShabiShett07 You'll have to update `evalrational.js`, and then run it to generate the rational functions, rather than copying and pasting them in `main.js`. You can refer: https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/bessely0/scripts/evalrational.js

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/bessely1/src/main.c:
```diff
@@ -0,0 +1,292 @@
+
+/**
+* Computes the Bessel function of the second kind of order one.
+*
```
```suggestion
*
* ## Notes
*
* -   Accuracy for subnormal `x` is very poor. Full accuracy is achieved at `1.0e-308` but trends progressively to zero at `5e-324`. This suggests that underflow (or overflow, perhaps due to a reciprocal) is effectively cutting off digits of precision until the computation loses all accuracy at `5e-324`.
*
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/bessely1/src/main.c:
```diff
@@ -0,0 +1,292 @@
+* @return     evaluated Bessel function
+*
+* @example
+* double v = y1(0.0);
```
```suggestion
* double v = y1( 0.0 );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/bessely1/src/main.c:
```diff
@@ -0,0 +1,292 @@
+* // returns -Infinity
+*/
+
+// Main function to compute the Bessel function of the second kind of order one
```
```suggestion
```
This doesn't seem necessary.

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/bessely1/src/main.c:
```diff
@@ -0,0 +1,292 @@
+*/
+
+// Main function to compute the Bessel function of the second kind of order one
+double stdlib_base_bessely1(double x) {
```
```suggestion
double stdlib_base_bessely1( double x ) {
```

gunjjoshi replied:
Missing spaces here and elsewhere in this file.

gunjjoshi replied:
Also, there is a mix of spaces and tabs in this file. Can you please reiterate it once again, @ShabiShett07?

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/bessely1/test/test.native.js:
```diff
@@ -0,0 +1,286 @@
+	x = subnormal.x;
+	for ( i = 0; i < x.length; i++ ) {
+		y = y1( x[i] );
+		t.equal( y, NINF, 'returns -Infinity' );
```
```suggestion
		t.equal( y, NINF, 'returns expected value' );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/bessely1/test/test.native.js:
```diff
@@ -0,0 +1,286 @@
+	for ( i = 0; i < 1000; i++ ) {
+		x = -( randu() * 100.0 ) - EPS;
+		v = y1( x );
+		t.equal( isnan( v ), true, 'returns NaN' );
```
```suggestion
		t.equal( isnan( v ), true, 'returns expected value' );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/bessely1/test/test.native.js:
```diff
@@ -0,0 +1,286 @@
+
+tape( 'the function returns `-Infinity` if provided `0`', function test( t ) {
+	var v = y1( 0.0 );
+	t.equal( v, NINF, 'returns -Infinity' );
```
```suggestion
	t.equal( v, NINF, 'returns expected value' );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/bessely1/test/test.native.js:
```diff
@@ -0,0 +1,286 @@
+
+tape( 'the function returns `NaN` if provided `NaN`', function test( t ) {
+	var v = y1( NaN );
+	t.equal( isnan( v ), true, 'returns NaN' );
```
```suggestion
	t.equal( isnan( v ), true, 'returns expected value' );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/bessely1/test/test.native.js:
```diff
@@ -0,0 +1,286 @@
+
+tape( 'the function returns `0.0` if provided `+infinity`', function test( t ) {
+	var v = y1( PINF );
+	t.equal( v, 0.0, 'returns 0.0' );
```
```suggestion
	t.equal( v, 0.0, 'returns expected value' );
```

gunjjoshi commented on lib/node_modules/@stdlib/math/base/special/bessely1/test/test.native.js:
```diff
@@ -0,0 +1,286 @@
+
+tape( 'the function returns `NaN` if provided `-infinity`', function test( t ) {
+	var v = y1( NINF );
+	t.equal( isnan( v ), true, 'returns NaN' );
```
```suggestion
	t.equal( isnan( v ), true, 'returns expected value' );
```

gunjjoshi committed:
Apply suggestions from code review

Signed-off-by: Gunj Joshi <gunjjoshi8372@gmail.com>

ShabiShett07 committed:
fix: update README.md

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
fix: update benchmark.native.js

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
fix: update benchmark.js

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
fix: update main.c

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 commented:
Hey, @gunjjoshi sir, Thank you for reviewing and I have tried to solve the errors which you have have pointed out, please do check

ShabiShett07 committed:
Merge branch 'develop' into feature/bessely1

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

ShabiShett07 committed:
fix: update benchmark.js

Signed-off-by: Shabareesh Shetty <139731143+ShabiShett07@users.noreply.github.com>

Planeshifter commented:
<!-- stdlib-bot-tracking-issue-closure-check -->
:warning: **Tracking Issue Closure Warning** :warning:

I noticed your PR description contains closing keywords ("Resolves", "Closes", or "Fixes") referencing a "Tracking Issue".

**Why this matters:**
Tracking issues should typically remain open until all related sub-issues are completed. GitHub automatically closes issues with such closing keywords when the PR is merged.

**Required action:**
Use "Progresses:" or "Ref:" instead to reference the tracking issue without automatically closing it.

Thank you for your contribution to the project!

ShabiShett07 committed:
chore: update implementation

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: missing_dependencies
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
remove: unnnecessary files

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feature/bessely1

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/bessely1/test/test.native.js:
```diff
@@ -0,0 +1,286 @@
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
```
```suggestion
tape( 'main export is a function', opts, function test( t ) {
```

Missing `opts`

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/bessely1/test/test.native.js:
```diff
@@ -0,0 +1,286 @@
+var abs = require( '@stdlib/math/base/special/abs' );
+var y1 = require( './../lib' );
+
+
```
```suggestion
// VARIABLES //

var y1 = tryRequire( resolve( __dirname, './../lib/native.js' ) );
var opts = {
	'skip': ( y1 instanceof Error )
};
```

This is missing.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/bessely1/src/main.c:
```diff
@@ -0,0 +1,295 @@
+	if (x == INFINITY) {
+		return 0.0;
+	}
+	xc = x;
```
@ShabiShett07, is there a reason why a copy of `x` is present? I don’t see either `x` or `xc` being modified in the code below.

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/bessely1/src/main.c:
```diff
@@ -0,0 +1,295 @@
+		return NAN;
+	}
+	if (x == 0.0) {
+		return -INFINITY;
```
```suggestion
		return STDLIB_CONSTANT_FLOAT64_NINF;
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/bessely1/src/main.c:
```diff
@@ -0,0 +1,295 @@
+	if (x == 0.0) {
+		return -INFINITY;
+	}
+	if (x == INFINITY) {
```
```suggestion
	if ( x == STDLIB_CONSTANT_FLOAT64_PINF ) {
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/bessely1/src/main.c:
```diff
@@ -0,0 +1,295 @@
+	double xc;
+
+	if (x < 0.0) {
+		return NAN;
```
```suggestion
	if ( x < 0.0 ) {
```

Negative spacing.

anandkaranubc replied:
```suggestion
		return 0.0 / 0.0; // NaN
```

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/bessely1/src/main.c:
```diff
@@ -0,0 +1,295 @@
+static const double x21 = 1.3900e+03;
+static const double x22 = -6.4592058648672279948e-06;
+
+/* Begin auto-generated functions. The following functions are auto-generated. Do not edit directly. */
```
These seem to have been written manually.

anandkaranubc committed:
docs: revert the changes in related section

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
bench: revert JS changes and native C benchmarks

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
docs: add JSDoc tags

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
feat: add polyval scripting for C

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
fix: apply suggestions from review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
test: apply suggestions from review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: update copyright years

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: update polyval files

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc reviewed (dismissed):
Good to go!

anandkaranubc committed:
docs: add missing `var` declarations in examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc commented on lib/node_modules/@stdlib/math/base/special/bessely1/lib/native.js:
```diff
@@ -0,0 +1,67 @@
+* // returns -Infinity
+*
+* @example
+* var v = y1( 1.0 );
```
```suggestion
* var v = y1( 1.0 );
```
